### PR TITLE
Add citation information

### LINKED ISSUE #644: Citation?
I'm drafting a paper that makes use of Skorch. Is there a citation you'd like me to use?

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
It adds a BibTeX entry to use for citation purposes to the docs.

**Reference issues/PRs**
This closes #644.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -29,6 +29,18 @@ allow for that.
 Overall, skorch aims at being as flexible as PyTorch while
 having a clean interface as sklearn.
 
+If you use Skorch, please use this BibTeX entry:
```
Comment (BenjaminBossan) [inline docs/index.rst:32]:
```suggestion
If you use skorch, please use this BibTeX entry:
```

Comment (stsievert) [reply docs/index.rst:32]: I’m inclined to use capital letters I think software libraries are proper nouns. But that’s a separate PR.

File: docs/index.rst [hunk 2]
```diff
@@ -29,6 +29,18 @@ allow for that.
 Overall, skorch aims at being as flexible as PyTorch while
 having a clean interface as sklearn.
 
+If you use skorch, please use this BibTeX entry:
+
+.. code:: bibtex
+
+   @manual{skorch,
+     author       = {Marian Tietz and Thomas J. Fan and Daniel Nouri and Benjamin Bossan and {skorch Developers}},
```
Comment (thomasjpfan) [inline docs/index.rst:37]: I would put Nouri and Bossan ahead of me on this list.

```suggestion
     author       = {Marian Tietz and Daniel Nouri and Benjamin Bossan and Thomas J. Fan and {skorch Developers}},
```

File: docs/index.rst (hunk 1 above)
Comment (BenjaminBossan) [reply docs/index.rst]: We consistently use lowercase "s" for skorch.

File: docs/index.rst (hunk 2 above)
Comment (BenjaminBossan) [reply docs/index.rst:37]: Thomas, you are too modest.

**Pull request merged**